cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -158,6 +167,9 @@ public void handle(WorkflowClient.Status status) {
                                            if (status.getCode() == WorkflowClient.Status.Code.NOT_FOUND) {
                                              responder.sendStatus(HttpResponseStatus.NOT_FOUND);
                                            } else if (status.getCode() == WorkflowClient.Status.Code.OK) {
+                                             // This uses responder.sendByteArray because status.getResult returns a
+                                             // json string, and responder.sendJson would need deserialization and
+                                             // serialization.
                                              responder.sendByteArray(HttpResponseStatus.OK,
                                                                      status.getResult().getBytes(),
```

Comment:
If getResult() returns a String, then we should not use getBytes() without a Charset. Or use Bytes.toBytes(String)